src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1310,7 +1310,7 @@ public static function encryptUsing($encrypter)
      */
     protected function castAttributeAsHashedString($key, $value)
     {
-        return Hash::needsRehash($value) ? Hash::make($value) : $value;
+        return $value !== null && password_get_info($value)['algo'] === null ? Hash::make($value) : $value;
```

Comment:
If we hard-code the password_get_info function here, aren't we limiting ourselves to a specific hashing implementation? In my opinion, it is the responsibility of the hashing driver to inform us whether the provided string value has been hashed or not.

However, let's see what @taylorotwell thinks.

Reply:
the `password_get_info` returns an `algo` value from one of the hashing constants in PHP https://www.php.net/manual/en/password.constants.php